### ISSUE TITLE
drracket has become useless -- it no longer makes repl transparent

### ISSUE THREAD

**Issue by racket-bug-submit** [samth/test-bugs#157]
###### Originally submitted by matthias <matthias at ccs dot neu dot edu> on: Tue Mar 13 12:04:01 -0400 2012

``` racket
This bug report reduces 12629 to a bug that doesn't involve TR. 

(define (f x) x)
click run 
edit the def window in a useless manner
click run again 

See this error message: 

define-values: cannot re-define a constant: f
```
##### Release:

```
5.2.900.1--2012-03-13(6b516d0/g)
```
##### Environment:

```
macosx "Darwin antarctica.ccs.neu.edu 11.3.0 Darwin Kernel Version 11.3.0: Thu Jan 12
 18:47:41 PST 2012; root:xnu-1699.24.23~1/RELEASE_X86_64 x86_64" (x86_64-macosx/3m)
 (get-display-depth) = 32
Human Language: english
(current-memory-use) 520963904
Links: (links) = (); (links #:user? #f) = (); (links #:root? #t) = (); (links #:user? #f
 #:root? #t) = ()


Collections:
("/Users/matthias/0Unison/collects/"
 (".DS_Store" "compiled" "date" "finance" "info-domain" "info.ss" "pdf.ss~" "session"
 "short" "testing" "tll-collects" "utils" "web"))
("/Users/matthias/Library/Racket/5.2.900.1/collects"
 (non-existent-path))
("/Users/matthias/plt/collects"
 (".DS_Store" ".gitignore" "2htdp" "algol60" "at-exp" "browser" "combinator-parser"
 "compiler" "config" "data" "datalog" "db" "defaults" "deinprogramm" "drracket" "drscheme"
 "dynext" "embedded-gui" "eopl" "errortrace" "ffi" "file" "framework" "frtime" "games"
 "graphics" "gui-debugger" "guibuilder" "handin-client" "handin-server" "help" "hierlist"
 "honu" "htdp" "html" "icons" "images" "info-domain" "lang" "launcher" "lazy"
 "macro-debugger" "make" "meta" "mred" "mrlib" "mysterx" "mzcom" "mzlib" "mzscheme" "net"
 "openssl" "parser-tools" "picturing-programs" "plai" "planet" "plot" "preprocessor"
 "profile" "r5rs" "r6rs" "racket" "racklog" "rackunit" "raco" "reader" "readline" "redex"
 "repo-time-stamp" "rnrs" "s-exp" "scheme" "schemeunit" "scribble" "scribblings"
 "scriblib" "setup" "sgl" "sirmail" "slatex" "slideshow" "srfi" "srpersist" "stepper"
 "string-constants" "swindle" "syntax" "syntax-color" "teachpack" "test-box-recovery"
 "test-engine" "tests" "tex2page" "texpict" "trace" !
 "typed" "typed-racket" "typed-scheme" "unstable" "version" "waterworld" "web-server"
 "wxme" "xml" "xrepl"))

Computer Language: (("Determine language from source") (#(#t print mixed-fraction-e #f #t
 debug) (default) #() "#lang racket" #t #t))
```
###### This bug was converted from Gnats bug [12630](http://bugs.racket-lang.org/old/12630).

[anon-submit; matthias at ccs dot neu dot edu]

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 11:01:56 -0600, mflatt at cs dot utah dot edu wrote:

``` racket
Sorry... I fixed this bug last night but forgot to push. I'll push the
repair now.

At Tue, 13 Mar 2012 12:04:02 -0400, matthias@ccs.neu.edu wrote:
> A new problem report is waiting at
>   http://bugs.racket-lang.org/query/?cmd=view&pr=12630
> 
> Reported by matthias for release: 5.2.900.1--2012-03-13(6b516d0/g)
> 
> *** Description:
> This bug report reduces 12629 to a bug that doesn't involve TR. 
> 
> (define (f x) x)
> click run 
> edit the def window in a useless manner
> click run again 
> 
> See this error message: 
> 
> define-values: cannot re-define a constant: f
> 
> *** How to repeat:
> 
> 
> *** Environment:
> macosx "Darwin antarctica.ccs.neu.edu 11.3.0 Darwin Kernel Version 11.3.0: Thu 
> Jan 12 18:47:41 PST 2012; root:xnu-1699.24.23~1/RELEASE_X86_64 x86_64" 
> (x86_64-macosx/3m) (get-display-depth) = 32
> Human Language: english
> (current-memory-use) 520963904
> Links: (links) = (); (links #:user? #f) = (); (links #:root? #t) = (); (links 
> #:user? #f #:root? #t) = ()
> 
> 
> Collections:
> ("/Users/matthias/0Unison/collects/"
>  (".DS_Store" "compiled" "date" "finance" "info-domain" "info.ss" "pdf.ss~" 
> "session" "short" "testing" "tll-collects" "utils" "web"))
> ("/Users/matthias/Library/Racket/5.2.900.1/collects"
>  (non-existent-path))
> ("/Users/matthias/plt/collects"
>  (".DS_Store" ".gitignore" "2htdp" "algol60" "at-exp" "browser" 
> "combinator-parser" "compiler" "config" "data" "datalog" "db" "defaults" 
> "deinprogramm" "drracket" "drscheme" "dynext" "embedded-gui" "eopl" 
> "errortrace" "ffi" "file" "framework" "frtime" "games" "graphics" 
> "gui-debugger" "guibuilder" "handin-client" "handin-server" "help" "hierlist" 
> "honu" "htdp" "html" "icons" "images" "info-domain" "lang" "launcher" "lazy" 
> "macro-debugger" "make" "meta" "mred" "mrlib" "mysterx" "mzcom" "mzlib" 
> "mzscheme" "net" "openssl" "parser-tools" "picturing-programs" "plai" "planet" 
> "plot" "preprocessor" "profile" "r5rs" "r6rs" "racket" "racklog" "rackunit" 
> "raco" "reader" "readline" "redex" "repo-time-stamp" "rnrs" "s-exp" "scheme" 
> "schemeunit" "scribble" "scribblings" "scriblib" "setup" "sgl" "sirmail" 
> "slatex" "slideshow" "srfi" "srpersist" "stepper" "string-constants" "swindle" 
> "syntax" "syntax-color" "teachpack" "test-box-recovery" "test-engine" "tests" 
> "tex2page" "texpict" "trace" !
>  "typed" "typed-racket" "typed-scheme" "unstable" "version" "waterworld" 
> "web-server" "wxme" "xml" "xrepl"))
> 
> Computer Language: (("Determine language from source") (#(#t print 
> mixed-fraction-e #f #t debug) (default) #() "#lang racket" #t #t))
```

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 11:09:00 -0600, mflatt at cs dot utah dot edu wrote:

``` racket
Actually, it looks like I did push after all. Do you still see this
problem with commit 36aa2c42d or later?

At Tue, 13 Mar 2012 11:01:56 -0600, Matthew Flatt wrote:
> Sorry... I fixed this bug last night but forgot to push. I'll push the
> repair now.
> 
> At Tue, 13 Mar 2012 12:04:02 -0400, matthias@ccs.neu.edu wrote:
> > A new problem report is waiting at
> >   http://bugs.racket-lang.org/query/?cmd=view&pr=12630
> > 
> > Reported by matthias for release: 5.2.900.1--2012-03-13(6b516d0/g)
> > 
> > *** Description:
> > This bug report reduces 12629 to a bug that doesn't involve TR. 
> > 
> > (define (f x) x)
> > click run 
> > edit the def window in a useless manner
> > click run again 
> > 
> > See this error message: 
> > 
> > define-values: cannot re-define a constant: f
> > 
> > *** How to repeat:
> > 
> > 
> > *** Environment:
> > macosx "Darwin antarctica.ccs.neu.edu 11.3.0 Darwin Kernel Version 11.3.0: 
> Thu 
> > Jan 12 18:47:41 PST 2012; root:xnu-1699.24.23~1/RELEASE_X86_64 x86_64" 
> > (x86_64-macosx/3m) (get-display-depth) = 32
> > Human Language: english
> > (current-memory-use) 520963904
> > Links: (links) = (); (links #:user? #f) = (); (links #:root? #t) = (); (links 
> > #:user? #f #:root? #t) = ()
> > 
> > 
> > Collections:
> > ("/Users/matthias/0Unison/collects/"
> >  (".DS_Store" "compiled" "date" "finance" "info-domain" "info.ss" "pdf.ss~" 
> > "session" "short" "testing" "tll-collects" "utils" "web"))
> > ("/Users/matthias/Library/Racket/5.2.900.1/collects"
> >  (non-existent-path))
> > ("/Users/matthias/plt/collects"
> >  (".DS_Store" ".gitignore" "2htdp" "algol60" "at-exp" "browser" 
> > "combinator-parser" "compiler" "config" "data" "datalog" "db" "defaults" 
> > "deinprogramm" "drracket" "drscheme" "dynext" "embedded-gui" "eopl" 
> > "errortrace" "ffi" "file" "framework" "frtime" "games" "graphics" 
> > "gui-debugger" "guibuilder" "handin-client" "handin-server" "help" "hierlist" 
> > "honu" "htdp" "html" "icons" "images" "info-domain" "lang" "launcher" "lazy" 
> > "macro-debugger" "make" "meta" "mred" "mrlib" "mysterx" "mzcom" "mzlib" 
> > "mzscheme" "net" "openssl" "parser-tools" "picturing-programs" "plai" 
> "planet" 
> > "plot" "preprocessor" "profile" "r5rs" "r6rs" "racket" "racklog" "rackunit" 
> > "raco" "reader" "readline" "redex" "repo-time-stamp" "rnrs" "s-exp" "scheme" 
> > "schemeunit" "scribble" "scribblings" "scriblib" "setup" "sgl" "sirmail" 
> > "slatex" "slideshow" "srfi" "srpersist" "stepper" "string-constants" 
> "swindle" 
> > "syntax" "syntax-color" "teachpack" "test-box-recovery" "test-engine" "tests" 
> > "tex2page" "texpict" "trace" !
> >  "typed" "typed-racket" "typed-scheme" "unstable" "version" "waterworld" 
> > "web-server" "wxme" "xml" "xrepl"))
> > 
> > Computer Language: (("Determine language from source") (#(#t print 
> > mixed-fraction-e #f #t debug) (default) #() "#lang racket" #t #t))
```

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 13:14:55 -0400, matthias at ccs dot neu dot edu wrote:

``` racket
The bug report I submitted was running on head as of this morning 8:00am. 

[[In the meantime, I have been trying to roll back to a version that doesn't
have the bug. Sadly, I ran into another bug with the version I had picked,
so now I am on my third attempt to get a working dr.]]

On Mar 13, 2012, at 1:09 PM, Matthew Flatt wrote:

> Actually, it looks like I did push after all. Do you still see this
> problem with commit 36aa2c42d or later?
> 
> At Tue, 13 Mar 2012 11:01:56 -0600, Matthew Flatt wrote:
>> Sorry... I fixed this bug last night but forgot to push. I'll push the
>> repair now.
>> 
>> At Tue, 13 Mar 2012 12:04:02 -0400, matthias@ccs.neu.edu wrote:
>>> A new problem report is waiting at
>>>  http://bugs.racket-lang.org/query/?cmd=view&pr=12630
>>> 
>>> Reported by matthias for release: 5.2.900.1--2012-03-13(6b516d0/g)
>>> 
>>> *** Description:
>>> This bug report reduces 12629 to a bug that doesn't involve TR. 
>>> 
>>> (define (f x) x)
>>> click run 
>>> edit the def window in a useless manner
>>> click run again 
>>> 
>>> See this error message: 
>>> 
>>> define-values: cannot re-define a constant: f
>>> 
>>> *** How to repeat:
>>> 
>>> 
>>> *** Environment:
>>> macosx "Darwin antarctica.ccs.neu.edu 11.3.0 Darwin Kernel Version 11.3.0: 
>> Thu 
>>> Jan 12 18:47:41 PST 2012; root:xnu-1699.24.23~1/RELEASE_X86_64 x86_64" 
>>> (x86_64-macosx/3m) (get-display-depth) = 32
>>> Human Language: english
>>> (current-memory-use) 520963904
>>> Links: (links) = (); (links #:user? #f) = (); (links #:root? #t) = (); (links 
>>> #:user? #f #:root? #t) = ()
>>> 
>>> 
>>> Collections:
>>> ("/Users/matthias/0Unison/collects/"
>>> (".DS_Store" "compiled" "date" "finance" "info-domain" "info.ss" "pdf.ss~" 
>>> "session" "short" "testing" "tll-collects" "utils" "web"))
>>> ("/Users/matthias/Library/Racket/5.2.900.1/collects"
>>> (non-existent-path))
>>> ("/Users/matthias/plt/collects"
>>> (".DS_Store" ".gitignore" "2htdp" "algol60" "at-exp" "browser" 
>>> "combinator-parser" "compiler" "config" "data" "datalog" "db" "defaults" 
>>> "deinprogramm" "drracket" "drscheme" "dynext" "embedded-gui" "eopl" 
>>> "errortrace" "ffi" "file" "framework" "frtime" "games" "graphics" 
>>> "gui-debugger" "guibuilder" "handin-client" "handin-server" "help" "hierlist" 
>>> "honu" "htdp" "html" "icons" "images" "info-domain" "lang" "launcher" "lazy" 
>>> "macro-debugger" "make" "meta" "mred" "mrlib" "mysterx" "mzcom" "mzlib" 
>>> "mzscheme" "net" "openssl" "parser-tools" "picturing-programs" "plai" 
>> "planet" 
>>> "plot" "preprocessor" "profile" "r5rs" "r6rs" "racket" "racklog" "rackunit" 
>>> "raco" "reader" "readline" "redex" "repo-time-stamp" "rnrs" "s-exp" "scheme" 
>>> "schemeunit" "scribble" "scribblings" "scriblib" "setup" "sgl" "sirmail" 
>>> "slatex" "slideshow" "srfi" "srpersist" "stepper" "string-constants" 
>> "swindle" 
>>> "syntax" "syntax-color" "teachpack" "test-box-recovery" "test-engine" "tests" 
>>> "tex2page" "texpict" "trace" !
>>> "typed" "typed-racket" "typed-scheme" "unstable" "version" "waterworld" 
>>> "web-server" "wxme" "xml" "xrepl"))
>>> 
>>> Computer Language: (("Determine language from source") (#(#t print 
>>> mixed-fraction-e #f #t debug) (default) #() "#lang racket" #t #t))
```

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 12:35:03 -0600, mflatt at cs dot utah dot edu wrote:

``` racket
I figured out my problem, and I can replicate the bug, now.

As of last night, it turns out that there were two bugs with the same
effect. I turned off "populate compiled directories" in DrRacket to fix
the lower-level bug, but I forgot to try turning the DrRacket option
back on afterward.

At Tue, 13 Mar 2012 13:14:55 -0400, Matthias Felleisen wrote:
> 
> 
> The bug report I submitted was running on head as of this morning 8:00am. 
> 
> [[In the meantime, I have been trying to roll back to a version that doesn't
> have the bug. Sadly, I ran into another bug with the version I had picked,
> so now I am on my third attempt to get a working dr.]]
> 
> 
> On Mar 13, 2012, at 1:09 PM, Matthew Flatt wrote:
> 
> > Actually, it looks like I did push after all. Do you still see this
> > problem with commit 36aa2c42d or later?
> > 
> > At Tue, 13 Mar 2012 11:01:56 -0600, Matthew Flatt wrote:
> >> Sorry... I fixed this bug last night but forgot to push. I'll push the
> >> repair now.
> >> 
> >> At Tue, 13 Mar 2012 12:04:02 -0400, matthias@ccs.neu.edu wrote:
> >>> A new problem report is waiting at
> >>>  http://bugs.racket-lang.org/query/?cmd=view&pr=12630
> >>> 
> >>> Reported by matthias for release: 5.2.900.1--2012-03-13(6b516d0/g)
> >>> 
> >>> *** Description:
> >>> This bug report reduces 12629 to a bug that doesn't involve TR. 
> >>> 
> >>> (define (f x) x)
> >>> click run 
> >>> edit the def window in a useless manner
> >>> click run again 
> >>> 
> >>> See this error message: 
> >>> 
> >>> define-values: cannot re-define a constant: f
> >>> 
> >>> *** How to repeat:
> >>> 
> >>> 
> >>> *** Environment:
> >>> macosx "Darwin antarctica.ccs.neu.edu 11.3.0 Darwin Kernel Version 11.3.0: 
> >> Thu 
> >>> Jan 12 18:47:41 PST 2012; root:xnu-1699.24.23~1/RELEASE_X86_64 x86_64" 
> >>> (x86_64-macosx/3m) (get-display-depth) = 32
> >>> Human Language: english
> >>> (current-memory-use) 520963904
> >>> Links: (links) = (); (links #:user? #f) = (); (links #:root? #t) = (); 
> (links 
> >>> #:user? #f #:root? #t) = ()
> >>> 
> >>> 
> >>> Collections:
> >>> ("/Users/matthias/0Unison/collects/"
> >>> (".DS_Store" "compiled" "date" "finance" "info-domain" "info.ss" "pdf.ss~" 
> >>> "session" "short" "testing" "tll-collects" "utils" "web"))
> >>> ("/Users/matthias/Library/Racket/5.2.900.1/collects"
> >>> (non-existent-path))
> >>> ("/Users/matthias/plt/collects"
> >>> (".DS_Store" ".gitignore" "2htdp" "algol60" "at-exp" "browser" 
> >>> "combinator-parser" "compiler" "config" "data" "datalog" "db" "defaults" 
> >>> "deinprogramm" "drracket" "drscheme" "dynext" "embedded-gui" "eopl" 
> >>> "errortrace" "ffi" "file" "framework" "frtime" "games" "graphics" 
> >>> "gui-debugger" "guibuilder" "handin-client" "handin-server" "help" 
> "hierlist" 
> >>> "honu" "htdp" "html" "icons" "images" "info-domain" "lang" "launcher" 
> "lazy" 
> >>> "macro-debugger" "make" "meta" "mred" "mrlib" "mysterx" "mzcom" "mzlib" 
> >>> "mzscheme" "net" "openssl" "parser-tools" "picturing-programs" "plai" 
> >> "planet" 
> >>> "plot" "preprocessor" "profile" "r5rs" "r6rs" "racket" "racklog" "rackunit" 
> >>> "raco" "reader" "readline" "redex" "repo-time-stamp" "rnrs" "s-exp" 
> "scheme" 
> >>> "schemeunit" "scribble" "scribblings" "scriblib" "setup" "sgl" "sirmail" 
> >>> "slatex" "slideshow" "srfi" "srpersist" "stepper" "string-constants" 
> >> "swindle" 
> >>> "syntax" "syntax-color" "teachpack" "test-box-recovery" "test-engine" 
> "tests" 
> >>> "tex2page" "texpict" "trace" !
> >>> "typed" "typed-racket" "typed-scheme" "unstable" "version" "waterworld" 
> >>> "web-server" "wxme" "xml" "xrepl"))
> >>> 
> >>> Computer Language: (("Determine language from source") (#(#t print 
> >>> mixed-fraction-e #f #t debug) (default) #() "#lang racket" #t #t))
```

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 14:35:57 -0400, matthias at ccs dot neu dot edu wrote:

``` racket
The program that kills drracket at the moment is this: 

> #lang racket
> 
> (define-signature doom^ ())
> (define-signature roll^ ())
> (define client@ (unit (import doom^) (export roll^) 42))

I can't reproduce the (define (f x) …) version that I had earlier. 

But I did notice that the commit is different: 

> Welcome to DrRacket, version 5.2.900.1--2012-03-13(4b18db1/g) [3m].
> Language: racket; memory limit: 512 MB.
> . define-values: cannot re-define a constant: client@
> > 

[[ This is after a complete total rebuild from the git repository. ]] 

On Mar 13, 2012, at 1:39 PM, Matthew Flatt wrote:
```

> I know that you rebuilt the Racket executable this morning (which is where the fix resides). Can you double-check that
> 
>  racket/racket3m -l drracket
> 
> in your build directory also misbehaves?
>

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 16:42:12 -0400, mflatt at racket-lang dot org assigned this bug to mflatt

A commit by mflatt@racket-lang.org has resolved this report
  http://git.racket-lang.org/plt/commit/415b1eabc4

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 16:42:12 -0400, mflatt at racket-lang dot org wrote:

A commit by mflatt@racket-lang.org has resolved this report: 415b1eabc4
